### ISSUE TITLE
ENOENT when installing packages on windows with relative local path

### ISSUE THREAD

**Issue by petemounce** [npm/npm#12027]
I have a piece of shared code I want to package, within a repo, then extract to its own node package. I'm using npm `3.7.3` and node `5.9.0`

So, I've created my package.json, added necessary dependencies, and then in the calling app added that package as a dependency via 

``` powershell
# note, ps allows / directory separator
npm install ../../shared/loggerfactory --save
```

Which has added `"loggerfactory": "file:///C:\\src\\je\\cc\\tranalytics\\shared\\loggerfactory"` to the `dependencies` list of the package.json for my calling app.

When I then run `npm install` within the calling app, I get the output at https://gist.github.com/petemounce/b0f6370868517354f6c0.

Note double `c:\` prefix on the path that npm is attempting to try to find.

I can work around this by adjusting the package.json so the dependency is `file:///src\\je\\cc\\tranalytics\\shared\\loggerfactory` (no drive suffix).

I cannot use a relative path in the dependency declaration (like `file:///..\\..\\shared\\loggerfactory` or `file://..\\..\\shared\\loggerfactory`) - same ENOENT.

It looks like npm assumes paths cannot be relative from the current working directory (on windows at least). This means I cannot proceed with my extract-to-module approach, because my filesystem on continuous integration will not match my workstation's filesystem (let alone my co-workers).

My company doesn't have its own internal npm registry, and I'd prefer to avoid (at this stage, but perhaps not later) creating a repo per package, since at the moment this is a spike into running nodejs to achieve serverless apps via AWS Lambda + API Gateway. I was hoping for a low impact approach to refactoring the spike.

**Comment by zkat** [comment]
Hey nice catch! I've slapped the bug label at it, and we'll try and take a look. It's also possible this'll just get fixed as part of #11444, if our existing test suite covered this case and was just failing, so keep an eye out!

**Comment by busches** [comment]
This appears to be fixed by #11444 as @zkat predicted, as installing a relative dependency and running `npm install` with `npm` version `3.10.8` installs the module correctly.
